otelhttp: export Event type for WithMessageEvents (#8153)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

- Add unmarshaling and validation for `CardinalityLimits` and `SpanLimits` to v1.0.0 model in `go.opentelemetry.io/contrib/otelconf`. (#8043)
- Add unmarshaling and validation for `BatchLogRecordProcessor`, `BatchSpanProcessor`, and `PeriodicMetricReader` to v1.0.0 model in `go.opentelemetry.io/contrib/otelconf`. (#8049)
- Add unmarshaling and validation for `TextMapPropagator` to v1.0.0 model in `go.opentelemetry.io/contrib/otelconf`. (#8052)

### Changed

- Improve performance by reducing allocations in the gRPC stats handler in `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc`. (#8035)

> - Export the Event type in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` for use with `WithMessageEvents`, replacing the previously unexported event type. (#8153)

Reviewer: ```suggestion
- Export the `ReadEvents` and `WriteEvents` values  in `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` so they can be used in `WithMessageEvents`. (#8153)
```
